public/index.js

```diff
@@ -0,0 +1,178 @@
+// TODO: I reverted back to not nesting everything to get it mostly working again. Tips?
+var output = [];
+// TODO: look into bubbling! (note to self)
+var buttons = document.getElementsByClassName( "calculator-btn" );
+
+for (i = 0; i < buttons.length; i++) {
+  buttons[i].addEventListener( "click", function() {
+    var value = this.getAttribute( "data-value" );
+
+    controller(value);
+  })
+}
+
+function clear() {
+  output.length = 0;
+  document.querySelector("#clear").innerText = "AC";
+  setView(0);
+}
+
+function percentage() {
+  return output.pop() / 100;
+}
+
+function posNeg() {
+  return output.pop() * -1;
+}
+
+function dot() {
+  return output.pop() + ".";
+}
+
+function multiply(position, value) {
+  return position === -1
+         ?
+         output[output.length - 2] * value : output[0] * output[2];
+}
+
+function divide(position, value) {
+  return position === -1
+         ?
+         output[output.length - 2] / value : output[0] / output[2];
+}
+
+function add() {
+  return parseInt(output[0]) + parseInt(output[2]);
+}
+
+function subtract() {
+  return parseInt(output[0]) - parseInt(output[2]);
+}
+
+function calculate(operator) {
+  var val;
+
+  switch (operator) {
+    case "*":
+      val = multiply();
+      break;
+    case "/":
+      val = divide();
+      break;
+    case "+":
+      val = add();
+      break;
+    case "-":
+      val = subtract();
+      break;
+    default:
+      break;
+  }
+
+  return val;
+}
+
+function calculateController() {
+  var operators = ["*", "/", "+", "-"];
+
+  if (output.length === 1) return output[0];
+
+  var operator = operators.indexOf(output[1]);
+
+  if (output.length === 2) output.push(output[0]);
+
+  return calculate(operators[operator]);
+  // does not yet account for 2 + 3 *
+}
+
+function setView(val) {
+  console.log(output);
+  document.querySelector("#calculator-output").innerText = val;
+}
+
+function controller(value) {
```

Comment:
this function is huge. try breaking it up into smaller, well named sub functions.